CMakeLists.txt

```
@@ -1867,11 +1867,25 @@ if(LILV)
 endif()
 
 # Live Broadcasting (Shoutcast)
-find_package(Shout)
-cmake_dependent_option(BROADCAST "Live Broadcasting (Shoutcast) support" ON "Shout_FOUND" OFF)
+option(BROADCAST "Live Broadcasting (Shoutcast) support" ON)
 if(BROADCAST)
-  if(NOT Shout_FOUND)
-      message(FATAL_ERROR "Live Broadcasting support requires the libshout and its development headers.")
+  if(UNIX AND NOT APPLE)
+    # Check if system lib is at least 2.4.4 and not suffering bug
+    # https://bugs.launchpad.net/mixxx/+bug/1833225
+    find_package(Shout 2.4.4)
+    if(Shout_FOUND AND Shout_VERSION VERSION_LESS 2.4.4)
+        message(STATUS "System installed version is suffering bug lp1833225")
+    endif()
+    if(NOT Shout_FOUND OR Shout_VERSION VERSION_LESS 2.4.4)
+      # Fall back to internal libraray in the lib tree
+      message(STATUS "Preparing internal Shout")
```

I think we can just detect libshout without specificing a version, and handle the version check afterwards.
```suggestion
    find_package(Shout)
    if(NOT Shout_FOUND)
        message(STATUS "System does not have libshout, falling back to built-in library.")
    elseif(Shout_VERSION VERSION_LESS 2.4.4)
        message(STATUS "Installed libshout version has bug lp1833225, falling back to built-in library.")
    endif()
    if(NOT Shout_FOUND OR Shout_VERSION VERSION_LESS 2.4.4)
      # Fall back to internal library in the lib/ directory
      message(STATUS "Using built-in libshout")
```

Then we can even move the find_package call out of the OS-specific cases, directly after `if(BROADCAST)

---

But isn't the package used then and we have two libshourts in the linker kall.

I am unsure here?

How is it done normally? 
Interestingly, FOUND is set even if I give the version to find_package. 

We can also not set the required flag, because this fails the build.

---

find_package only finds the package library paths + include dirs, but doesn't add anything to the linker stage.
That is done here:
https://github.com/mixxxdj/mixxx/blob/bb1baff347609c4b2b13060b8260dbb256330d32/CMakeLists.txt#L1897

I didn't notice before, but you never actually link `shout_mixxx`. It always links `Shout::Shout`, which is system's libshout.

Also, is the problem only present on Linux/BSD or why does the code not apply for OSX/Windows as well?

---

> Also, is the problem only present on Linux/BSD or why does the code not apply for OSX/Windows as well?

I don't know, the bug was Linux only.
The other targets shouldn't be effected, because they do not depends on the system libraries.

---

But if we have libshout in the mixxx source tree anyway, we could also fallback to that if libshout is not found, instead of throwing an error and aborting compliation on Windows/OSX.

---

Yes of cause. But than we need to write a full features CMakeList for libshout.

This is of a low prio, compared to the fix here. So I like to postpone it.

---

We are planning switch to libshout-idjc for AAC support by the way.